### PR TITLE
Chore/docs/make single landing page

### PR DESCRIPTION
**Pull request by ZhukovGreen**
This MR centralizes contribution guidelines into one place.

Content moved to https://leapp.readthedocs.io/en/latest/contributing.html contributing.md from several sources:
- https://github.com/oamg/leapp-guidelines
- https://github.com/oamg/leapp-repository/blob/master/CONTRIBUTING.md

## TODO
- [ ] squash commits after review
- [x] Update/remove definition of done

### CODE REVIEW

File: docs/source/best-practices.md [hunk 1]
```diff
@@ -4,6 +4,13 @@
 
 See the [contribution guidelines for leapp-repository](contributing).
```
Comment (bocekm) [inline docs/source/best-practices.md:5]:
```suggestion
See the [contribution guidelines](contributing).
```

File: docs/source/best-practices.md [hunk 2]
```diff
@@ -4,6 +4,13 @@
...
 
+## Naming convention
```
Comment (bocekm) [inline docs/source/best-practices.md:7]: This is IMHO not a best practice. In best practices we shouldn't use the word `must`. Instead, best practices should contain stuff that's not mandatory but highly recommended.
I'd put it to the `docs/source/contributing.md` under a section `File naming convention`.

File: docs/source/best-practices.md (hunk 1 above)
Comment (bocekm) [reply docs/source/best-practices.md:5]: The contribution guidelines are shared for both leapp and leapp-repository.

File: docs/source/contributing.md [hunk 3]
```diff
@@ -1,14 +1,30 @@
...
+
+## Best practices in actor development
+
+- actor development: [Best practices for actor development](best-practices.html)
+- actor directory layout might be found in 
+[Repository directory layout](repository-dir-layout.html)
+(under `actorname/`)
+- actor testing: [Testing actors](test-actors.html)
+
+## RPM packages naming schema
```
Comment (bocekm) [inline docs/source/contributing.md:22]: I think that the way how are the Copr RPMs named does not relate to contributions to the project from others much. I'd maybe link to that `build-schema` page only from under https://leapp.readthedocs.io/en/latest/infrastructure.html.

File: docs/source/contributing.md [hunk 4]
```diff
@@ -1,14 +1,30 @@
...
 The following is a set of guidelines for contributing effectively to the Leapp-related repositories
 hosted under the `OS and Application Modernization Group organization <https://github.com/oamg/>`_
 on GitHub.
 
-Submitting a Pull Request
-^^^^^^^^^^^^^^^^^^^^^^^^^
+## Code style guidelines
+
+Your code should follow our
+[Python Coding Guidelines](python-coding-guidelines.html)
```
Comment (bocekm) [inline docs/source/contributing.md:12]:
```suggestion
[Python Coding Guidelines](python-coding-guidelines)
```
Do not link to `.html`, there was a problem with that. See https://github.com/oamg/leapp/pull/602. Update the other references as well.

File: docs/source/test-actors.md [hunk 5]
```diff
@@ -25,3 +33,7 @@ The Leapp actors are covered by three types of tests - unit, component and e2e.
 
 - The Leapp QA team maintains an internal testing framework facilitating e2e tests.
 - [Members of the *oamg* GitHub organization] can trigger execution of the e2e tests by adding the comment 'e2e tests' under an opened *leapp* or *leapp-repository* PR
+
+## Bug fixing of actor
+
+Each bug found and fixed in the actor, should be covered by tests
```
Comment (bocekm) [inline docs/source/test-actors.md:39]:
```suggestion
Each bug found and fixed in an actor should be covered by tests.
```

File: docs/source/test-actors.md [hunk 6]
```diff
@@ -8,6 +8,14 @@ The Leapp actors are covered by three types of tests - unit, component and e2e.
 - Unit and component tests modules should have unique names
 - Tutorial on [How to write unit and component tests](unit-testing)
 
+## Naming conventions
+
+Test modules names should match the following regex:
```
Comment (bocekm) [inline docs/source/test-actors.md:13]:
```suggestion
Test module names should match the following regex:
```

File: docs/source/python-coding-guidelines.md [hunk 7]
```diff
@@ -1,5 +1,178 @@
...
+msg_many_args = "{who} {when} {what}".format(who="A Lanister", when="always", what="pays his debts")
+```
+
+### 12. Docstrings
+
+Follow [PEP 257 - Docstring Conventions](https://www.python.org/dev/peps/pep-0257)
+  - with the exception, that the summary line of a multi-line docstring shall be on a new line, not on the same line as the opening quotes.
+
+There are some minimal requirements for the information that actor docstrings should provide, to learn more about the specifics please
+consult [leapp-repository contributing guidelines](https://github.com/oamg/leapp-repository/blob/master/CONTRIBUTING.md).
```
Comment (bocekm) [inline docs/source/python-coding-guidelines.md:139]:
```suggestion
consult the [contribution guidelines](contributing).
```

File: README.md [hunk 8]
```diff
@@ -5,7 +5,9 @@ for more infomation and documenation.**
 
 ## Contributing
 If you are looking for information about contributing into this project, kindly
```
Comment (bocekm) [inline README.md:7]:
```suggestion
If you are looking for information about contributing to this project, kindly
```

File: docs/source/definition-of-done.md [hunk 9]
```diff
@@ -0,0 +1,12 @@
+# Definiton of done
+
+ * [ ] The acceptance criteria described in the Task is met
```
Comment (bocekm) [inline docs/source/definition-of-done.md:3]:
```suggestion
 * [ ] The acceptance criteria described in the Jira Issue are met
```
1 criterion, 2+ criteria

File: docs/source/definition-of-done.md [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+ * [ ] The code is approved and merged into the repository(ies)
+ * [ ] The documentation is updated
+   * Official landing page: [](https://leapp.readthedocs.io/en/latest/contributing.html)
```
Comment (bocekm) [inline docs/source/definition-of-done.md:6]:
```suggestion
```
The contribution guidelines is not a landing page for Leapp as project.

File: docs/source/definition-of-done.md [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+   * Developers documentation: https://leapp.readthedocs.io/en/latest/
```
Comment (bocekm) [inline docs/source/definition-of-done.md:7]:
```suggestion
   * Upstream documentation: https://leapp.readthedocs.io/en/latest/
```

File: docs/source/definition-of-done.md [hunk 12]
```diff
@@ -0,0 +1,12 @@
...
+   * Presentation for external contributors: https://goo.gl/Foh2tk
+ * [ ] Dev. Images are updated if necessary
```
Comment (bocekm) [inline docs/source/definition-of-done.md:9]:
```suggestion
 * [ ] Development vagrant box is updated if necessary
```

File: docs/source/definition-of-done.md [hunk 13]
```diff
@@ -0,0 +1,12 @@
+# Definiton of done
```
Comment (bocekm) [inline docs/source/definition-of-done.md:1]:
```suggestion
# Definition of done for internal Jira Issues
```

This definition of done is so old, internal and not used at all. I would rather remove it completely from the upstream project documentation and mention it for example in https://docs.google.com/document/d/1T_HQ1HoyjLlHcdNGlKp_D17Qs3s5strR0Y_y2mg8rBs/.

Do @GAZDOWN, @pirat89, @drehak and @ZhukovGreen agree?

File: docs/source/definition-of-done.md [hunk 14]
```diff
@@ -0,0 +1,12 @@
...
+ * [ ] The acceptance criteria described in the Task is met
+ * [ ] The code is approved and merged into the repository(ies)
+ * [ ] The documentation is updated
+   * Official landing page: [](https://leapp.readthedocs.io/en/latest/contributing.html)
+   * Developers documentation: https://leapp.readthedocs.io/en/latest/
+   * Presentation for external contributors: https://goo.gl/Foh2tk
+ * [ ] Dev. Images are updated if necessary
+ * [ ] Unit tests that cover the new code are written and passing
+ * [ ] Integration tests are passing
+ * [ ] The ticket contains link(s) with the results of the Task (e.g., PR, document, etc.).
```
Comment (bocekm) [inline docs/source/definition-of-done.md:12]:
```suggestion
 * [ ] The Jira Issue contains link(s) to the results (e.g., PR, document, etc.).
```

File: docs/source/contributing.md [hunk 15]
```diff
@@ -1,14 +1,30 @@
...
+
+## Best practices in actor development
+
+- actor development: [Best practices for actor development](best-practices.html)
```
Comment (bocekm) [inline docs/source/contributing.md:16]:
```suggestion
* Actor development: [Best practices for actor development](best-practices.html)
```
For the sake of consistency (to have it the same as in the rest of the file).

File: docs/source/contributing.md [hunk 16]
```diff
@@ -1,14 +1,30 @@
...
+- actor directory layout might be found in 
+[Repository directory layout](repository-dir-layout.html)
+(under `actorname/`)
```
Comment (bocekm) [inline docs/source/contributing.md:19]:
```suggestion
* Actor directory layout is to be found at [Repository directory layout](repository-dir-layout.html) (under `actorname/`)
```

File: docs/source/definition-of-done.md (hunk 13 above)
Comment (ZhukovGreen) [reply docs/source/definition-of-done.md:1]: I agree. This doesn't make sense to me. Our interanl stuff should remains internal.

File: docs/source/best-practices.md (hunk 2 above)
Comment (ZhukovGreen) [reply docs/source/best-practices.md:7]: I think this Naming convention is more confusing then helpful. 
- Why do we require do not use underscores? 
- Then requirements to the test modules we have in testing actors doc. 
- Do not use actor name in modules sounds like common sense and we don't need to write common sense derived things here, otherwise, we could never end.
- And requirement to the naming of a main module of actor actor.py is looking like mandatory and should probably stay in best-practices for actor development. 

I suggest keeping the actor.py requirement inside best-practices and this is it. The rest should be skipped.

Comment (bocekm) [reply docs/source/best-practices.md:7]: > I think this Naming convention is more confusing then helpful.
> 
>     * Why do we require do not use underscores?
To have the naming consistent, it's as easy as that. We didn't want one actor be named `checkonething` and the other `check_the_other_thing`. I'd not remove this one.
> 
>     * Then requirements to the test modules we have in testing actors doc.
I don't understand this bullet.
> 
>     * Do not use actor name in modules sounds like common sense and we don't need to write common sense derived things here, otherwise, we could never end.
It may seem like a common sense now because you see now tens of actors and models that follow this rule. But it was not like that from the beginning. I agree that it can be dropped now because there's already enough examples to give you the idea of naming.
> 
>     * And requirement to the naming of a main module of actor actor.py is looking like mandatory and should probably stay in best-practices for actor development.
For mandatory requirements in documentation, word `shall` is being used instead of `must`. And as I wrote, I'd like to avoid having mandatory requirements in best practices. It sounds like having word `must` in a meal recipe. This is not a best practice, it's a requirement mandated by the leapp framework.
> 
> 
> I suggest keeping the actor.py requirement inside best-practices and this is it. The rest should be skipped.

Comment (pirat89) [reply docs/source/best-practices.md:7]: I think the naming convention should really be part of contribution guidelines. Best practices tells, you doesn't have to do it neccessary. But in this case, we are talking about something what will be required. As well, I think that is not needed to put there any duplicated info about name of actor script `actor.py` - that's the requirement from the framework already, so no need to document that twice.

> Why do we require do not use underscores?

long story short, we use (or is it just planned?) underscores to be able to create uniq key for the actor based on the path in the repository.

Comment (ZhukovGreen) [reply docs/source/best-practices.md:7]: > * Then requirements to the test modules we have in testing actors doc.
I meant that how to test modules should be named already described in 
`test-actors.md`.

The requirement about the naming of `actor.py` is also here: docs/source/repository-dir-layout.md. I don't see any other reference from the leapp framework. I would keep it in contributing.md


So should I go with:
```diff
--- a/docs/source/contributing.md
+++ b/docs/source/contributing.md
@@ -15,7 +15,14 @@ Your code should follow our
 
 * Actor development: [Best practices for actor development](best-practices)
 * Actor directory layout is to be found at [Repository directory layout](repository-dir-layout) (under `actorname/`)
-- actor testing: [Testing actors](test-actors.html)
+* Actor testing: [Testing actors](test-actors.html)
+
+## File naming convention
+
+1. New folders and/or Python files shall use lowercase without underscores.
+1. Do not use _actor_ in the actor's name, neither _model/topic/tag/phase_ 
+in the model/topic/tag/phase name, as these things are implicit.
+1. The actor's main file shall be named `actor.py`.
```
?

File: docs/source/definition-of-done.md (hunk 13 above)
Comment (ZhukovGreen) [reply docs/source/definition-of-done.md:1]: ping  @GAZDOWN, @pirat89, @drehak :) I am about to remove it. Do you agree?

File: docs/source/definition-of-done.md (hunk 9 above)
Comment (ZhukovGreen) [reply docs/source/definition-of-done.md:3]: waiting for a decision regarding existency of definition of done

File: docs/source/definition-of-done.md (hunk 10 above)
Comment (ZhukovGreen) [reply docs/source/definition-of-done.md:6]: waiting for a decision regarding existency of definition of done

File: docs/source/definition-of-done.md (hunk 11 above)
Comment (ZhukovGreen) [reply docs/source/definition-of-done.md:7]: waiting for a decision regarding existency of definition of done

File: docs/source/definition-of-done.md (hunk 14 above)
Comment (ZhukovGreen) [reply docs/source/definition-of-done.md:12]: waiting for a decision regarding existency of definition of done

File: docs/source/index.rst [hunk 17]
```diff
@@ -20,6 +18,7 @@ Resources not included in this RTD:
     inplace-upgrade-workflow
     contributing
     python-coding-guidelines
+    definition-of-done
```
Comment (ZhukovGreen) [inline docs/source/index.rst:21]: waiting for a decision regarding existency of definition of done

File: docs/source/definition-of-done.md (hunk 12 above)
Comment (ZhukovGreen) [reply docs/source/definition-of-done.md:9]: waiting for a decision regarding existency of definition of done

**Comment by ZhukovGreen** [comment]
@bocekm I think were are ready to merge

File: docs/source/contributing.md [hunk 18]
```diff
@@ -1,14 +1,28 @@
...
+## Code style guidelines
+
+Your code should follow our
+[Python Coding Guidelines](python-coding-guidelines)
+
+## Best practices in actor development
+
+* Actor development: [Best practices for actor development](best-practices)
+* Actor directory layout is to be found at [Repository directory layout](repository-dir-layout) (under `actorname/`)
+* Actor testing: [Testing actors](test-actors.html)
```
Comment (bocekm) [inline docs/source/contributing.md:18]:
```suggestion
* Actor testing: [Testing actors](test-actors)
```

**Comment by ZhukovGreen** [comment]
@bocekm after the merge we might also archive https://github.com/oamg/leapp-guidelines right?
